superset/datasource/api.py

```diff
@@ -116,8 +116,14 @@ def get_column_values(
 
         row_limit = apply_max_row_limit(app.config["FILTER_SELECT_ROW_LIMIT"])
         try:
+            # always denormalize column name before querying for values
+            db_engine_spec = datasource.database.db_engine_spec
+            db_dialect = datasource.database.get_dialect()
+            denomalized_col_name = db_engine_spec.denormalize_name(
+                db_dialect, column_name
+            )
```

Comment:
It's better to move this logic to the `datasource.values_for_column` method, so that the API doesn't have any business logic, and so that calls to `datasource.values_for_column` from other places also work as expected.